src/utilities/utility.ts

```diff
@@ -129,6 +131,93 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
     return commandExecutionResults;
 }
 
+export async function getFilePathsConfigs(): Promise<any> {
+
+    let filePathsConfig: any = {};
+    const MANIFEST_PATHS_KEY = 'manifestFilePaths';
+    const HELM_CHART_KEY = 'helmChartFilePaths';
+    const DOCKERFILE_PATH_KEY = 'dockerfilePaths';
+    const DOCKERFILE_PATH_LABEL_KEY = 'dockerfile-path';
+
+    let inputManifestFiles = inputParams.manifests || [];
+    filePathsConfig[MANIFEST_PATHS_KEY] = inputManifestFiles;
+
+    let helmChartPaths = (process.env.HELM_CHART_PATHS && process.env.HELM_CHART_PATHS.split('\n').filter(path => path != "")) || [];
+    filePathsConfig[HELM_CHART_KEY] = helmChartPaths;
+
+    //Parsing dockerfile paths for images
+    let imageNames = core.getInput('images').split('\n');
+    let imageDockerfilePathMap: any = {};
+    let pathKey: any, pathVal: any;
+
+    //Fetching from env var if available
+    let dockerfilePathsList: any[] = (process.env.DOCKERFILE_PATHS &&  process.env.DOCKERFILE_PATHS.split('\n')) || [];
+    dockerfilePathsList.forEach(path => {
+        if(path){
+            pathKey = path.split(' ')[0];
+            pathVal = path.split(' ')[1];
+            imageDockerfilePathMap[pathKey] = pathVal;
+        }
+    })
+
+    //Fetching from image lable if available
+    for(const image of imageNames){
+        let args: string[] = [image];
+        let resultObj: any;
+        let containerRegistryName = image;
+
+        try{
+            let usrname = process.env.CR_USERNAME || null;
+            let pwd = process.env.CR_PASSWORD || null;
+            if(pwd && usrname)
+            {
+                let loginArgs: string[] = [containerRegistryName, '--username', usrname, '--password', pwd];
+                await exec.exec('docker login ', loginArgs, true).then(res => {
+                    if (res.stderr != '' && !res.success) {
+                        throw new Error(`docker login failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                    }
+                });
+            }
+
+            await exec.exec('docker pull ', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker images pull failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+            });
+
+            await exec.exec('docker inspect --type=image', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker inspect call failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+
+                if(res.stdout){
+                    resultObj = JSON.parse(res.stdout);
+                }
+            });   
+        }

```

Comment:
How are you handling a multi image scenario?

Reply:
It is not handled here.
I will make the env variable an array of values that would be parsed, to handle the multiple images from multiple registries.

Reply:
Added the design change for multiple registries-credentials and multiple image-dockerfile paths.